include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -211,7 +211,23 @@ __joint_inclusive_scan(_Args&&... __args)
 }
 
 #if _ONEDPL_FPGA_DEVICE
-#    if _ONEDPL_LIBSYCL_VERSION >= 50300
+#    if _ONEDPL_LIBSYCL_VERSION >= 60100
+struct __fpga_emulator_selector
+{
+    int operator()(const sycl::device& d) const
+    {
+        return sycl::ext::intel::fpga_emulator_selector_v(d);
+    }
+};
```

Comment:
It seems it relies on implementation details because [the extension specification](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_fpga_device_selector.asciidoc) does not describe a constructor, which has a device as an argument. Selector constructed using a device puzzles me, I expected the opposite... 

What do you think about changing it to:
```c++
// include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h
inline auto __fpga_emulator_selector()
{
    return sycl::ext::intel::fpga_emulator_selector_v;
}
inline auto __fpga_selector()
{
    return sycl::ext::intel::fpga_selector_v;
}

#    elif _ONEDPL_LIBSYCL_VERSION >= 50300
inline auto __fpga_emulator_selector()
{
    return sycl::ext::intel::fpga_emulator_selector{};
}
inline auto __fpga_selector()
{
    return sycl::ext::intel::fpga_selector{};
}
.....

// include\oneapi\dpl\pstl\hetero\dpcpp\execution_sycl_defs.h
    fpga_policy()
        : base(sycl::queue(
#    if _ONEDPL_FPGA_EMU
              __dpl_sycl::__fpga_emulator_selector()
#    else
              __dpl_sycl::__fpga_selector()
#    endif // _ONEDPL_FPGA_EMU
```

Reply:
This is not calling a constructor; the SYCL 2020 selectors are actually free-standing functions that take a `sycl::device` as input and return an `int`. Although, it is confusing the way I have written it.

I pushed an updated version based on your suggestion.

Reply:
Thanks for the clarification. This is what the specification say in https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-selector.